src/customers/service/customer.service.ts

```diff
@@ -0,0 +1,89 @@
+import * as bcrypt from 'bcryptjs';
+
+import {
+  HttpException,
+  HttpStatus,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
+
+import { Customer } from '../domain/customer.entity';
+import { CustomerRepository } from '../repository/customer.respository';
+
+@Injectable()
+export class CustomerService {
+  constructor(private readonly customerRepository: CustomerRepository) {}
+
+  async getAll(): Promise<Customer[]> {
+    return this.customerRepository.findAll();
+  }
+
+  async getById(id: string): Promise<Customer> {
+    const customer = await this.customerRepository.findOneById(id);
+    if (!customer) {
+      throw new NotFoundException('No customer found by the given ID');
+    }
+    return customer;
+  }
+
+  async getByEmailAddress(emailAddress: string): Promise<Customer> {
+    const customer =
+      await this.customerRepository.findByEmailAddress(emailAddress);
+    if (!customer) {
+      throw new NotFoundException('No customer found by the given email');
+    }
+
+    return customer;
+  }
+
+  async isEmailAvailable(emailAddress: string): Promise<boolean> {
+    const customer =
+      await this.customerRepository.findByEmailAddress(emailAddress);
+
+    return customer ? false : true;
+  }
+
+  async create(newCustomer: Omit<Customer, 'id' | 'createdAt'>): Promise<void> {
+    const isEmailAvailable = await this.isEmailAvailable(
+      newCustomer.emailAddress,
+    );
+    if (!isEmailAvailable) {
+      throw new HttpException(
```

Comment:
**suggestion:** if you want to throw customized exceptions you should add and build them in a separate place (with their unique message) and later throw them directly here